### PR TITLE
#662: finished ticket, users now properly sorted

### LINKED ISSUE #662: Login Page - Organize the Users on the Dev Login List
### Description

Currently the login dropdown for the development environment sorts the seed users by user id. However, this is not an intuitive way to display the names.

We want to sort the names in a better way. Organize the users on the log-in list so that they're grouped by role and sorted by name as indicated below.

### Acceptance Criteria

Names should be sorted as follows: (see mock below)

1. role (all app admins should come first, then all admins, then leadership, members, guests)
2. last name (ascending alphabetically, if the roles are the same)
3. first name (ascending alphabetically, if the roles and last name are the same)
4. user id (ascending order, if everything else is the same which honestly shouldn't happen)

### Proposed Solution

Find the dev login box component. Figure out where the names are ordered and change the comparator so that it follows the desired ordering. You may want to refresh yourself on how comparator functions work. 

### Mocks

IMG_7548
Screen Shot 2023-01-24 at 10 15 22 PM

### PR DESCRIPTION
**Pull request by andrewstsai**
## Changes

Users on dev login lists now sorted depending on their role, last name, first name, and user id with decreasing priority

## Notes

Copied over rankUserRole to shared folder so it could be used for frontend

## Test Cases

- Case A
- Edge case
- ...

## Screenshots
 
pr

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #662

### CODE REVIEW

File: src/shared/src/rank-user-role.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      case 'ADMIN':
+        return 4;
+      case 'LEADERSHIP':
+        return 3;
+      case 'MEMBER':
+        return 2;
+      default:
+        return 1;
+    }
+  };
```
Comment (Peyton-McKee) [inline src/shared/src/rank-user-role.ts:17]: need new line here.

File: src/frontend/src/pages/LoginPage/LoginDev.tsx [hunk 2]
```diff
@@ -44,6 +45,9 @@ const LoginDev: React.FC<LoginDevProps> = ({ devSetUser, devFormSubmit }) => {
         >
           {usersList
             .sort((a, b) => a.userId - b.userId)
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/LoginPage/LoginDev.tsx:47]: no need to sort by userId.

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
Great Work! In addition to these comments you need to delete the rankUserRole from the backend and instead whereever it was called use the shared function

**Review by Peyton-McKee — APPROVED** [review]
LGTM

**Review by anthonybernardi — APPROVED** [review]
very clean!

**Pull request merged**